smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/middleware/MiddlewareExecutionGenerator.kt

```diff
@@ -51,25 +51,11 @@ class MiddlewareExecutionGenerator(
             SmithyHTTPAPITypes.HTTPRequest,
             SmithyHTTPAPITypes.HTTPResponse,
         )
-        writer.write(
-            """
-            config.interceptorProviders.forEach { provider in
-                builder.interceptors.add(provider.create())
-            }
-            """.trimIndent()
-        )
-        writer.openBlock(
-            "config.httpInterceptorProviders.forEach { (provider: any \$N) -> Void in",
-            "}",
-            ClientRuntimeTypes.Core.HttpInterceptorProvider,
-        ) {
-            writer.write(
-                "let i: any \$N<\$N, \$N> = provider.create()",
-                ClientRuntimeTypes.Core.HttpInterceptor,
-                inputShape,
-                outputShape,
-            )
-            writer.write("builder.interceptors.add(i)")
```

Comment:
Since the return value of `provider.create()` now matches the type of `interceptors.add()` exactly, there is no need to assign to a temp variable with an explicit type anymore.

Also:
- The writer statements for non-HTTP interceptor providers has been rewritten from a multiline statement.
- The block explicit type added previously in an attempt to address compiler crashes is removed since it seems to have had no effect.